base/reduce.jl

```diff
@@ -243,10 +233,12 @@ pairwise_blocksize(::typeof(abs2), ::typeof(+)) = 4096
 _empty_reduce_error() = throw(ArgumentError("reducing over an empty collection is not allowed"))
 mr_empty(f, op, T) = _empty_reduce_error()
 # use zero(T)::T to improve type information when zero(T) is not defined
-mr_empty(::typeof(identity), op::typeof(+), T) = r_promote(op, zero(T)::T)
-mr_empty(::typeof(abs), op::typeof(+), T) = r_promote(op, abs(zero(T)::T))
-mr_empty(::typeof(abs2), op::typeof(+), T) = r_promote(op, abs2(zero(T)::T))
-mr_empty(::typeof(identity), op::typeof(*), T) = r_promote(op, one(T)::T)
+mr_empty(::typeof(identity), op::typeof(+), T) = zero(T)::T
+mr_empty(::typeof(abs), op::typeof(+), T) = abs(zero(T)::T)
+mr_empty(::typeof(abs2), op::typeof(+), T) = abs2(zero(T)::T)
+mr_empty(::typeof(identity), op::typeof(*), T) = one(T)::T
```

Comment:
This is incorrect, since `one(T)` may not be of type `T` for a dimensionful type.  Just get rid of the typeassert here.